### PR TITLE
Remove a bunch of API surface from core

### PR DESCRIPTION
**Pull request by adamcik**
Note that the original goal was just #1070 - but I've done a bit more. As such I've left the changelog out for now as we might want to rebase this dropping some of these changes...

### CODE REVIEW

File: mopidy/core/playback.py [hunk 1]
```diff
@@ -267,20 +266,17 @@ def pause(self):
...
         Play the given track, or if the given track is :class:`None`, play the
         currently active track.
 
         :param tl_track: track to play
         :type tl_track: :class:`mopidy.models.TlTrack` or :class:`None`
-        :param on_error_step: direction to step at play error, 1 for next
-            track (default), -1 for previous track. **INTERNAL**
-        :type on_error_step: int, -1 or 1
         """
+        self._play(tl_track, 1)
```
Comment (jodal) [inline mopidy/core/playback.py]: Would be more readable with `self._play(tl_track, on_error_step=1)`.

**Comment by jodal** [comment]
+1, changelog please :-)

**Comment by adamcik** [comment]
This time I wasn't being lazy, I was postponing it as I wasn't sure if we wanted to merge all of these :-)

**Comment by jodal** [comment]
I know, and I think they all look good.

**Comment by jodal** [comment]
Ah, no hint about review comments being fixed!

**Pull request merged**